force-app/main/default/classes/ServiceSessionSelector.cls

```diff
@@ -15,6 +15,42 @@ public with sharing class ServiceSessionSelector {
     @TestVisible
     private FieldBucketSelector bucketSelector = new FieldBucketSelector();
 
+    public List<ServiceSession__c> getServiceSessionsForServiceSchedule(
+        Id serviceScheduleId
+    ) {
+        List<ServiceSession__c> serviceSessions = new List<ServiceSession__c>();
+
+        if (!Schema.SObjectType.ServiceSession__c.isAccessible()) {
+            return serviceSessions;
+        }
+
+        Set<String> fields = new Set<String>();
+        fields.add(String.valueOf(ServiceSession__c.Id)); // Name is required
+        fields.add(String.valueOf(ServiceSession__c.Name)); // Name is required
+        fields.add(String.valueOf(ServiceSession__c.Status__c)); // Status is required
+        fields.add(String.valueOf(ServiceSession__c.SessionStart__c)); // Start date is used to group the records
+        fields.add(String.valueOf(ServiceSession__c.SessionEnd__c)); // Start date is used to group the records
+        fields.add(String.valueOf(ServiceSession__c.ServiceSchedule__c)); // Start date is used to group the records
+
+        QueryBuilder queryBuilder = new QueryBuilder()
```

Comment:
If we know all of the fields ahead of time that we are querying for, do we need to use the query builder or can we just use SELECT...?